Analytics/importers/scheduler_new.py

```diff
@@ -7,105 +7,141 @@
 import sqlalchemy
 from sqlalchemy.orm import sessionmaker, scoped_session
 from apscheduler.schedulers.background import BackgroundScheduler
-from apscheduler.executors.pool import ProcessPoolExecutor, ThreadPoolExecutor
+from apscheduler.jobstores.sqlalchemy import SQLAlchemyJobStore
 
 import settings
+from importers.state_decorator import ImporterStatus
+from models.importer_status import ImporterStatuses
+from app import create_app
+from db import db
 
+application = create_app()
 sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
 logging.basicConfig(level='INFO', filename='importers.log', filemode='a')
 logger = logging.getLogger(__name__)
 
+engine = sqlalchemy.create_engine(settings.DB_URI)
+session = scoped_session(sessionmaker(bind=engine))
+jobstore = {
+    'sqlalchemy': SQLAlchemyJobStore(url=settings.DB_URI)
+}
+sched = BackgroundScheduler(jobstores=jobstore)
+sched.start()
+
 
 class Scheduler(object):
     """ Schedule the execution importer tasks """
 
-    def __init__(self):
-        """ Create SQLAlchemy session and launch APScheduler """
-
-        self.engine = sqlalchemy.create_engine(settings.DB_URI)
-        self.session = scoped_session(sessionmaker(bind=self.engine))
+    status_tracker = ImporterStatus.get_importer_status()
 
-        executor = {
-            'default': ThreadPoolExecutor(10),
-            'processpool': ProcessPoolExecutor(5)
-        }
-        self.scheduler = BackgroundScheduler(executors=executor)
-        self.scheduler.start()
-
-    def get_apis(self) -> list:
+    @status_tracker.changed.register
+    def status_has_changed(self, status: ImporterStatuses):
+        """
+        Receive the status of an importer and persist it to the Importer
+        Status table
+        :param status: Status object identifying the
+        :return:
```

Comment:
Remove since not used

Reply:
The self parameter is required for the @status_tracker decorator to function correctly. If it is removed, the corresponding import class name will not be returned